airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py

```diff
@@ -176,12 +176,15 @@ def parse_response(self, response: requests.Response, stream_state: Mapping[str,
             return []
 
         for gzip_filename in zip_file.namelist():
-            with zip_file.open(gzip_filename) as file:
-                for record in self._parse_zip_file(file):
-                    if record[self.cursor_field] >= state_value:
-                        user_id_key = f"{self.name}_user_id" # to avoid name collision with user_id field
-                        record[user_id_key] = record.pop("user_id")
-                        yield self._date_time_to_rfc3339(record)  # transform all `date-time` to RFC3339
+            try:
+                with zip_file.open(gzip_filename) as file:
```

Comment:
@a-rampalli In the case where it is failing to open zip file, won't adding try/catch lead to data loss?

Reply:
Yes. But based on my understanding this error is caused due to faulty response from API even though we get 200 in response. Any suggestions on how we can avoid data loss?

Reply:
API response for https://amplitude.com/api/2/export\?start\=20220907T20\&end\=20220908T19

![Screenshot 2023-02-08 at 2 47 34 PM](https://user-images.githubusercontent.com/110388330/217487120-b1106b0b-755b-4f12-8328-b04edb816344.png)

Reply:
This API response doesn't indicate a failure, AFAIR this API returns a zip file, which is supposed to be non-readable.

Reply:
Yes. Turns out actual issue is caused due to not handling null date-time values in api response. Fixed it now.